include/inl/utilities.inl

```diff
@@ -0,0 +1,87 @@
+//
+// include/utilities.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#include "../utilities.hpp"
+
+namespace gnc {
+namespace utl {
+
+template <typename T>
+constexpr void quat_conj(lin::Vector<T, 4> const &q, lin::Vector<T, 4> &res) {
+  res = { -q(0), -q(1), -q(2), q(3) };
+}
+
+template <typename T>
+constexpr void quat_conj(lin::Vector<T, 4> &q) {
+  quat_conj(q, q);
+}
+
+template <typename T>
+constexpr void quat_cross_mult(Vector4f const &q1, Vector4f const &q2, Vector4f &res) {
+  auto const qv1 = lin::ref<3, 1>(q1, 0, 0);
+  auto const qv2 = lin::ref<3, 1>(q2, 0, 0);
+  lin::ref<3, 1>(res, 0, 0) = lin::cross(qv2, qv1) + q1(3) * qv2 + q2(3) * qv1;
+  res(3) = q1(3) * q2(4) - lin::dot(qv1, qv2);
```

Comment:
should be `q2(3)` instead of `q2(4)`  Matlab 1 based indexing is the worst.

Reply:
You are totally right lol